osquery/tables/system/linux/pci_devices.cpp

```diff
@@ -26,38 +31,162 @@ const std::string kPCIKeyVendor = "ID_VENDOR_FROM_DATABASE";
 const std::string kPCIKeyModel = "ID_MODEL_FROM_DATABASE";
 const std::string kPCIKeyID = "PCI_ID";
 const std::string kPCIKeyDriver = "DRIVER";
+const std::string kPCISubsysID = "PCI_SUBSYS_ID";
+
+const std::string kPciIdsPath = "/usr/share/misc/pci.ids";
+
+PciDB::PciDB(std::istream& db_filestream) {
+  // pci.ids keep track of subsystem information of vendor and models
+  // sequentially so we keep track of what the current vendor and models are.
+  std::string cur_vendor, cur_model, line;
+  while (std::getline(db_filestream, line)) {
+    if (line.size() < 7 || line.at(0) == '\n' || line.at(0) == '#') {
+      continue;
+    }
+
+    switch (line.find_first_of("0123456789abcdef")) {
+    case 0: {
+      // Vendor info.
+      cur_vendor = line.substr(0, 4);
+
+      // Once we get to illege vendor section we can stop since we're not
+      // currently parsing device classes.
+      if (cur_vendor == "ffff") {
+        return;
+      }
+
+      // Setup current vendor.
+      auto vendor = PciVendor();
+      vendor.id = cur_vendor;
+      // Bump 2 chars to account for whitespace separation..
+      vendor.name = line.substr(6);
+      db_[cur_vendor] = vendor;
+
+      break;
+    }
+
+    case 1:
+      // Model info.
+      if (db_.find(cur_vendor) != db_.end() && line.size() > 7) {
+        cur_model = line.substr(1, 4);
+
+        // Set up current model under the current vendor.
+        auto model = PciModel();
+        model.id = cur_model;
+        // Bump 2 chars to account for whitespace separation.
+        model.desc = line.substr(7);
+        db_[cur_vendor].models[cur_model] = model;
```

Comment:
Don't you need to check if cur_vendor is not empty first?

Reply:
The check occurs at line `70`